pkg/security/podsecuritypolicy/provider.go

```diff
@@ -310,6 +316,10 @@ func (s *simpleProvider) ValidateContainerSecurityContext(pod *api.Pod, containe
 		}
 	}
 
+	if !s.psp.Spec.AllowPrivilegeEscalation && sc.AllowPrivilegeEscalation != nil && *sc.AllowPrivilegeEscalation {
```

Comment:
I think this always happens after defaulting, which means we should require that `AllowPrivilegeEscalation != nil`. Otherwise if the admin forgets to set a default and the user leaves this empty, that's equivalent to allowing privilige escalation in certain scenarios (IIUC). Alternatively, we could default to false if it's prohibited by the psp and no default is set.